packages/jsii-rosetta/lib/tablets/tablets.ts

```diff
@@ -121,8 +130,17 @@ export class LanguageTablet {
     return this.snippets[snippetKey(typeScriptSource)];
   }
 
+  /**
+   * Load the tablet from a file. Will automatically detect if the file is in
+   * compression format ('.gz') and gunzip if it is.
+   */
   public async load(filename: string) {
-    const obj = (await fs.readJson(filename, { encoding: 'utf-8' })) as TabletSchema;
+    let obj: TabletSchema | any = {};
+    if (path.extname(filename) === '.gz') {
+      obj = JSON.parse(zlib.gunzipSync(await fs.readFile(filename)).toString()) as TabletSchema;
+    } else {
+      obj = (await fs.readJson(filename, { encoding: 'utf-8' })) as TabletSchema;
+    }
```

Comment:
Instead of using extension detection -- since we know to expect either a Gzip object or a JSON document, we can actually use the header of the gzip object, which starts with a known magic number:

```suggestion
    let data = await fs.readFile(filename);
    // Gzip objects start with 1F 8B 08
    if (data[0] === 0x1F && data[1] === 0x8B && data[2] === 0x08) {
      // This is a gz object, so we decompress it now...
      data = lib.gunzipSync(data);
    }
    const obj: TabletSchema = JSON.parse(data.toString('utf-8'));
```

Reply:
Fascinating stuff. The more you know...